src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -242,6 +257,10 @@ object RocksDbDataSource {
   case object IterationFinished extends RuntimeException
   case class IterationError(ex: Throwable)
 
+  class RocksDbDataSourceClosedException(val message: String) extends IllegalStateException(message)
```

Comment:
what about reason field? It's usually very helpful in debugging if the real issue lies a layer below